pyspark/bigdl/nn/layer.py

```diff
@@ -929,6 +929,40 @@ class FeedForwardNetwork(Layer):
     def __init__(self, hidden_size, filter_size, relu_dropout, bigdl_type="float"):
         super(FeedForwardNetwork, self).__init__(None, bigdl_type,
                                         hidden_size, filter_size, relu_dropout)
+class LayerNormalization(Layer):
+    '''
+    Applies layer normalization.
+
+    >>> norm = LayerNormalization(8)
+    creating: createLayerNormalization
+    '''
+
+    def __init__(self, hidden_size, bigdl_type="float"):
+        super(LayerNormalization, self).__init__(None, bigdl_type, hidden_size)
+
+class TableOperationExpand(Layer):
+    '''
+    When two tensors have different size, firstly expand small size tensor to large size tensor,
+    and then do table operation.
+
+    >>> norm = TableOperationExpand(CMulTable())
+    creating: createCMulTable
+    creating: createTableOperationExpand
+    '''
+
+    def __init__(self, operation_layer, expand_pos=2, bigdl_type="float"):
```

Comment:
why we should specify the expand_pos ? can we infer from input ? it's determined by the input stead of layer itself right ?

Reply:
yeah, good idea. Refine code and infer position from input